src/appsettings/settings.py

```diff
@@ -367,15 +369,23 @@ class Setting(object):
     the variable is missing and the setting is not required. If the setting
     is missing and required, trying to access it will raise an AttributeError.
 
-    When accessing a setting's value, the value is first fetched from the
-    project settings, then passed to a transform function that will return it
-    modified (or not). By default, no transformation is applied.
+    When accessing a setting's value, the value is first fetched from environment
+    or the project settings, then passed to a transform function that will return
+    it modified (or not). By default, no transformation is applied.
 
     The call_default parameter tells if we should try to call the given
     default value before returning it. This allows to give lambdas or callables
     as default values. The transform_default parameter tells if we should
     transform the default value as well through the transform method.
 
+    If the setting value is taken from environment, decode_environ method is
+    called. By default it just decodes string as JSON and throws JSONDecodeError
+    if the value is not a valid JSON. Some subclasses override and extend this
+    method to be able to handle even other common values.
+
+    Environment variables are not passed to inner settings of ``NestedDictSetting``
+    and ``NestedListSetting``.
```

Comment:
This block is probably way to descriptive for the class docstring and it's very probably we will forgot to maintain it in here. I'd keep the description of the internals in their own docstrings.

Reply:
I think it should be mentioned even here, but I see your point, moved this to the subclasses.